### PR TITLE
feat(frontend): move wallet connect btn out of menu into header

### PR DESCRIPTION
**Pull request by loki344**
# Motivation

Instead of displaying the wallet connect button as a menu item in the popover, we want to display it in the header at any time.

# Changes

- remove MenuItem for WalletConnect out of dropdown to the header
- extend WalletConnectButton to show connected and disconnected state.
- rename existing MenuWallet into MenuAdresses because that's the only content now

# Tests

before: 
image

after:
image

image

### CODE REVIEW

File: src/frontend/src/lib/components/core/Menu.svelte [hunk 1]
```diff
@@ -56,8 +56,8 @@
 
 <Popover bind:visible anchor={button} direction="rtl">
 	<div class="flex flex-col gap-4" data-tid={NAVIGATION_MENU}>
-		{#if walletOptions}
-			<MenuWallet on:icMenuClick={hidePopover} />
+		{#if yourAddressesOption}
+			<MenuYourAddresses on:icMenuClick={hidePopover} />
```
Comment (loki344) [inline src/frontend/src/lib/components/core/Menu.svelte:60]: Out of curiosity, is there a good reason to not show this option when a user is in the settings route?

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/core/Menu.svelte:60]: that is a good question... I remember there was some talk about this at some point, but the PR is https://github.com/dfinity/oisy-wallet/pull/1900

Maybe worth asking design team?

File: src/frontend/src/lib/components/hero/Header.svelte [hunk 2]
```diff
@@ -40,6 +41,7 @@
 		{/if}
 
 		{#if $authSignedIn}
+			<WalletConnectMenu />
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/hero/Header.svelte:44]: doe we keep the other WalletConnect button that is above this lines?? Or one exclude the other?

File: src/frontend/src/lib/components/hero/WalletConnectMenu.svelte [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		});
+	};
+</script>
+
+<ButtonIcon
+	ariaLabel={$i18n.wallet_connect.text.name}
+	on:click={openWalletConnectAuth}
+	disabled={$walletConnectPaired}
+>
+	<IconWalletConnect size="22" slot="icon" />
```
[the remark this answers is not in the document]
Comment (loki344) [reply src/frontend/src/lib/components/hero/WalletConnectMenu.svelte:24]: I added the label as suggested

File: src/frontend/src/lib/components/core/MenuYourAddresses.svelte
[the remark this answers is not in the document]
Comment (loki344) [reply src/frontend/src/lib/components/core/MenuYourAddresses.svelte:1]: I'm going with `MenuAddresses `

File: src/frontend/src/lib/components/hero/Header.svelte (hunk 2 above)
Comment (loki344) [reply src/frontend/src/lib/components/hero/Header.svelte:44]: You mean the `<WalletConnect/>`? This seems to be a wrapper around different WalletConnectModals, so yes I leave it.

File: src/frontend/src/lib/components/core/Menu.svelte (hunk 1 above)
Comment (loki344) [reply src/frontend/src/lib/components/core/Menu.svelte:60]: Perfect, thank you for linking this issue - It seems as the YourAdresses functionality also does not work on the /explore route, so I have to exclude it there as well.

File: src/frontend/src/lib/components/hero/Header.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/hero/Header.svelte:44]: that button appears when an user is connected to WalletConnect; so if it's connected, we would have 2 wallet connect button?

File: src/frontend/src/lib/components/hero/WalletConnectMenu.svelte [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	<svelte:fragment>{$i18n.wallet_connect.text.name}</svelte:fragment>
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/hero/WalletConnectMenu.svelte:25]: no need for `svelte:fragment` here

File: src/frontend/src/lib/components/hero/Header.svelte (hunk 2 above)
Comment (loki344) [reply src/frontend/src/lib/components/hero/Header.svelte:44]: Ah you're completely right. Now I see what you meant

File: src/frontend/src/eth/components/wallet-connect/WalletConnectSession.svelte [hunk 5]
```diff
@@ -336,12 +338,23 @@
...
+			name: TRACK_COUNT_WALLET_CONNECT_MENU_OPEN
+		});
+	};
 </script>
 
 {#if nonNullish(listener)}
 	<WalletConnectButton on:click={disconnect}
 		>{$i18n.wallet_connect.text.disconnect}</WalletConnectButton
 	>
+{:else}
```
Comment (loki344) [inline src/frontend/src/eth/components/wallet-connect/WalletConnectSession.svelte:355]: @AntonioVentilii-DFINITY How do you test the wallet connect functionality locally?

Comment (peterpeterparker) [reply src/frontend/src/eth/components/wallet-connect/WalletConnectSession.svelte:355]: I generally connect with https://app.uniswap.org/

File: src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte [hunk 6]
```diff
@@ -1,9 +1,17 @@
 <script lang="ts">
 	import IconWalletConnect from '$lib/components/icons/IconWalletConnect.svelte';
 	import { ethAddressNotLoaded } from '$lib/derived/address.derived';
+
+	export let ariaLabel: string | undefined = undefined;
 </script>
 
-<button on:click class="tertiary text-primary h-10" disabled={$ethAddressNotLoaded}>
+<button
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte:8]: maybe some transition, to make the connection change smoother?

File: src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte [hunk 7]
```diff
@@ -1,9 +1,17 @@
...
+	aria-label={ariaLabel}
+	on:click
+	class="tertiary text-primary h-10"
+	class:icon={!$$slots.default}
```
[the remark this answers is not in the document]
Comment (loki344) [reply src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte:12]: Yes that is correct.

File: src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte (hunk 6 above)
Comment (loki344) [reply src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte:8]: Can you help me out here?

What transition were you thinking of? I think if we want to have a smooth transition, I would have to change the usage in WalletConnectSession.svelte to render the same button, but with different content / onClick handlers?

Beginning of the video i just switch between connected / disconnected to show the transition fade from svelte.
End of the video is the actual implementation with the connect modal.

https://github.com/user-attachments/assets/a85cc334-5a08-40a5-bc0e-706fb9784aee

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte:8]: I would first try one thing:

what if the button is just one and you change the props/labels based on a condition?

```svelte
<WalletConnectButton
	on:click={nonNullish(listener) ? disconnect : openWalletConnectAuth}
	ariaLabel={nonNullish(listener) ? $i18n.wallet_connect.text.name : the_other_arialabel}
>
	{#if nonNullish(listener)}
		<span transition:slide={SLIDE_PARAMS}>{$i18n.wallet_connect.text.disconnect}</span>
	{/if}
</WalletConnectButton>
```


I am not 100% sure if it is better or not, though, or if it fits the requirements

Comment (loki344) [reply src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte:8]: Had to tried different things and ended up with:
https://github.com/user-attachments/assets/fa33f871-e012-4e0b-9761-7a0a47caf139

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte:9]: thank you!

**Review by AntonioVentilii-DFINITY — APPROVED** [review]

**Pull request merged**